installation/cross_compile/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Prerequisites
+Please make sure everything is ready for you to proceed:
+- Raspberry Pi OS ready on your Raspberry Pi ([tutorial](https://www.raspberrypi.org/documentation/installation/installing-images/)).
+- SSH access to Raspberry Pi ([tutorial](../README.md#wifi-and-ssh)).
+- Docker on your PC ([tutorial](https://docs.docker.com/get-docker/)).
+- Optional SSH server on your PC (`sudo apt install openssh-server`).
+
+## Raspberry Pi Preparation
+
+ROS2 requires the following packages to be installed on your Raspberry Pi:
+```bash
+sudo apt update
+
+# Compilation dependencies
+sudo apt install \
+    liblog4cxx-dev \
+    python3-dev
+
+# Runtime dependencies
+sudo apt install \
+    python3-numpy \
+    python3-netifaces \
+    python3-yaml
+
+# Optional tools
+sudo apt install sshfs
+```
+
+## ROS2 Cross-Compilation on Your PC
+
+Now we can start a process of ROS2 cross-compilation.
+The supplied Docker image is equipped with most of the tools you need, so start it with:
+```bash
+./start_docker.sh
+```
+This command will build and run the Docker container, and allocate a pseudo-TTY.
+It means that you can use it as an another operating system.
+From now on, you will be able to do most of the tasks from the Docker container.
+
+
+### Rootfs Preparation
+
+As your Raspberry Pi is equipped with ROS2 dependencies you have to synchronize the [rootfs](https://wiki.dlang.org/GDC/Cross_Compiler/Existing_Sysroot#:~:text=A%20sysroot%20is%20a%20folder,sysroot%2Fusr%2Finclude'.).
+This will allow the cross-compiler to use header files and libraries from your Raspberry Pi.
+There are two ways to get the rootfs inside your Docker.
+
+The first way, you can synchronize the content of the rootfs:
+```bash
+rsync -rLR --safe-links pi@[raspberry_pi_ip]:/{lib,usr,opt/vc/lib} /home/develop/rootfs
+```
+
+> Initially, `rsync` will take more time to perform synchronization, but the cross-compilation process will be faster after (as the cross-compiler doesn't have to transfer a file from Raspberry Pi every time).
+
+
+The second way, you can use `sshfs` tool to mount the rootfs:
+```bash
+sshfs -o follow_symlinks,allow_other -o cache_timeout=115200 pi@[raspberry_pi_ip]:/ /home/develop/rootfs
+```
+
+### Compilation Commands 
+
+In the Docker, we prepared a few commands with prefix `cross-*` to bootstrap your development.
+For example, you can use `cross-initialize` to download ROS2 source code or `cross-colcon-build` to build it.
+These commands are simple bash functions located in `.bashrc`.
+You can see how the commands are implemented by typing e.g. `type cross-initialize` and change them according to your needs.
+Therefore, in the Docker container type:
+```bash
+cross-initialize
+```
+to download ROS2 source code.
+By default, it will initialize ROS2 Foxy distribution, but if you want to change it you can use `ROS_DISTRO` environment variable, e.g. for ROS2 Dashing:
```

Comment:
```suggestion
By default, it will initialize the ROS2 Foxy distribution, but if you want to change it you can use `ROS_DISTRO` environment variable, e.g. for ROS2 Dashing:
```